### PR TITLE
Correct the Syntax of IUnaryOperation (logical not) in control flow g…

### PR DESCRIPTION
**Pull request by usr-sse2**
…raph

In the current release, the IOperation control flow graph for a method:
```cs
void f() {
    var x = !g(); 
}
```
contains an `IUnaryOperation` for the logical not, but its `Syntax` is `InvocationExpressionSyntax` `g()` instead of `PrefixUnaryExpressionSyntax` `!g()`.

### CODE REVIEW

**Comment by jcouv** [comment]
@usr-sse2 Thanks for fixing this, but we didn't hear back from you on adding corresponding test.
I'll mark the PR as "personal" for now. Please ping once the feedback was addressed and it is ready for review again.

**Comment by usr-sse2** [comment]
@jcouv @333fred I added the test, on my computer it passes with my previous commit and doesn't pass without it.

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IReturnStatement.cs [hunk 1]
```diff
@@ -421,6 +421,42 @@ int F(bool a)
...
+
+            string expectedGraph = @"
+Block[B0] - Entry
+    Statements (0)
+    Next (Regular) Block[B1]
+Block[B1] - Block
+    Predecessors: [B0]
+    Statements (0)
+    Next (Return) Block[B2]
+        IUnaryOperation (UnaryOperatorKind.Not) (OperationKind.UnaryOperator, Type: System.Boolean) (Syntax: '!f')
```
Comment (333fred) [inline src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IReturnStatement.cs]:
```suggestion
        IUnaryOperation (UnaryOperatorKind.Not) (OperationKind.Unary, Type: System.Boolean) (Syntax: '!f')
```

Looks like you need to switch the OperationKind text being used here for the standardized one, this was changed in a recent commit. It should be fine with the suggested change.

**Comment by AlekseyTs** [comment]
It looks like there are legitimate test failures

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IReturnStatement.cs [hunk 2]
```diff
@@ -421,6 +421,42 @@ int F(bool a)
             VerifyFlowGraphForTest<BlockSyntax>(compilation, expectedGraph);
         }
 
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void ReturnFlow_UnaryNegation()
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IReturnStatement.cs]: >ReturnFlow_UnaryNegation [](start = 20, length = 24)

I think this test really belongs in IOperationTests_IUnaryOperatorExpression.cs. It looks like we don't have a test covering ```!!f``` scenario, please add one, I expect to not see a IUnaryOperation in the flow graph for it. Also, please add similar tests for VB.

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 2)

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IReturnStatement.cs (hunk 2 above)
Comment (usr-sse2) [reply src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IReturnStatement.cs]: c96b2be60a45fe1f73df97f27ebaedc7d635f7c2 9d525ebc29b3ff5848c87645e2ed412f4e65c3a9

**Comment by usr-sse2** [comment]
Test failures are the same as in master

**Comment by 333fred** [comment]
@AlekseyTs please take another look.

**Comment by 333fred** [comment]
@usr-sse2 thanks for the contribution!

**Pull request merged**